### PR TITLE
Fix quadratic indexing

### PR DESCRIPTION
**Pull request by joaquimg**

### CODE REVIEW

File: src/LinQuadOptInterface.jl [hunk 1]
```diff
@@ -113,7 +113,8 @@ ConstraintMapping() = ConstraintMapping(
 This function updates all of the references in `m`
 after we have deleted row `row` in the affine constraint matrix.
 """
-function shift_references_after_delete_affine!(m, row)
+shift_references_after_delete_affine!(m, row) = shift_references_after_delete_affine_base!(m, row)
+function shift_references_after_delete_affine_base!(m, row)
```
Comment (odow) [inline src/LinQuadOptInterface.jl]: Can you add a comment about why this is needed. This confused me until I went and looked at your Xpress branch.

File: src/solver_interface.jl [hunk 2]
```diff
@@ -142,6 +142,14 @@ Get the number of linear constraints in the model `m`.
 function get_number_linear_constraints end
 @deprecate lqs_getnumrows get_number_linear_constraints
 
+"""
+    get_last_linear_constraint_index(m)::Int
+
+Get the index of the last linear constraints added to the model `m`.
+"""
+function get_last_linear_constraint_index end
+get_last_linear_constraint_index(m) = get_number_linear_constraints(m)
```
Comment (odow) [inline src/solver_interface.jl]: Ditto with comments. Something along the lines of users should only implement this if the indexing is different

File: src/objective.jl [hunk 3]
```diff
@@ -131,7 +131,7 @@ function MOI.get(m::LinQuadOptimizer, ::MOI.ObjectiveFunction{Quad})
         for j in nzrange(Q, i)
             row = rows[j]
             val = vals[j]
-            push!(quadratic_terms, MOI.ScalarQuadraticTerm{Float64}(0.5*val, m.variable_references[row], m.variable_references[i]))
+            push!(quadratic_terms, MOI.ScalarQuadraticTerm{Float64}(val, m.variable_references[row], m.variable_references[i]))
```
Comment (odow) [inline src/objective.jl]: We need tests to catch this

File: src/LinQuadOptInterface.jl (hunk 1 above)
Comment (joaquimg) [reply src/LinQuadOptInterface.jl]: I will remove this!

File: src/objective.jl (hunk 3 above)
Comment (joaquimg) [reply src/objective.jl]: Yes! I am not even sure that is the right way

**Comment by joaquimg** [comment]
Update coming!
Xpress will have its internal mapping!

**Comment by joaquimg** [comment]
Quadratic indexing was modified directly in Xpress interface with no effect in LQOI

**Pull request closed without merging**